### PR TITLE
dug 0.0.94 (new formula)

### PR DESCRIPTION
**Pull request by awfulwoman**
This PR newly adds the dug DNS propagation checker.

- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

File: Formula/d/dug.rb [hunk 1]
```diff
@@ -0,0 +1,23 @@
+class Dug < Formula
+  desc "Global DNS propagation checker that gives pretty output"
+  homepage "https://dug.unfrl.com"
+  url "https://github.com/unfrl/dug/archive/refs/tags/0.0.93.tar.gz"
```
Comment (chenrui333) [inline Formula/d/dug.rb:4]: this tag did not get marked as latest release yet

File: Formula/d/dug.rb [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+  sha256 "a375034bcad3193d47a485b3dd33c7633f7890e5f79b6a9b6b3b3bc198fb2ec3"
+  license "MIT"
+
+  depends_on "dotnet"
+  uses_from_macos "zlib"
+
+  def install
+    ENV["DOTNET_ROOT"] = "$HOMEBREW_PREFIX/opt/dotnet/libexec"
```
Comment (chenrui333) [inline Formula/d/dug.rb:12]:
```suggestion
    ENV["DOTNET_ROOT"] = libexec
```

Comment (awfulwoman) [reply Formula/d/dug.rb]: Removed. I think this was just due to trying to debug the usage of dotnet on llinux.

File: Formula/d/dug.rb (hunk 1 above)
Comment (awfulwoman) [reply Formula/d/dug.rb:4]: Thanks for raising this in the original repo!

File: Formula/d/dug.rb [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+  sha256 "a375034bcad3193d47a485b3dd33c7633f7890e5f79b6a9b6b3b3bc198fb2ec3"
+  license "MIT"
+
+  depends_on "dotnet"
+  uses_from_macos "zlib"
+
+  def install
+    ENV["DOTNET_SYSTEM_GLOBALIZATION_INVARIANT"] = "1"
+    system "dotnet", "publish", "-c", "Release", "-p:PublishSingleFile=true",
+            "-p:IncludeNativeLibrariesForSelfExtract=true", "--self-contained",
```
Comment (chenrui333) [inline Formula/d/dug.rb:14]: should be --no-self-contained mode?

File: Formula/d/dug.rb [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+            "true", "-o", "publish", "./cli"
+    bin.install "./publish/dug"
```
Comment (chenrui333) [inline Formula/d/dug.rb:16]: might be better to build like other dotnet formulae

```
    dotnet = Formula["dotnet"]
    os = OS.mac? ? "osx" : OS.kernel_name.downcase
    arch = Hardware::CPU.intel? ? "x64" : Hardware::CPU.arch.to_s

    args = %W[
      --configuration Release
      --framework net#{dotnet.version.major_minor}
      --output #{libexec}
      --runtime #{os}-#{arch}
      --no-self-contained
      -p:PublishSingleFile=true
      -p:IncludeNativeLibrariesForSelfExtract=true
    ]

    system "dotnet", "publish", "cli/dug.csproj", *args
    env = { DOTNET_ROOT: "${DOTNET_ROOT:-#{dotnet.opt_libexec}}" }
    (bin/"dug").write_env_script libexec/"dug", env
```

File: Formula/d/dug.rb (hunk 2 above)
Comment (chenrui333) [reply Formula/d/dug.rb]: sounds good

File: Formula/d/dug.rb (hunk 3 above)
Comment (awfulwoman) [reply Formula/d/dug.rb:14]: ah, I see what you other comment means now. `write_env_script` sets an envvar for an installed bin. Gotcha.

File: Formula/d/dug.rb (hunk 4 above)
Comment (awfulwoman) [reply Formula/d/dug.rb]: Nice.

**Comment by awfulwoman** [comment]
Urgh, things are failing now after the suggestions from @chenrui333. Sadly I won't have time now to look at this for a few days. Sorry.

**Comment by chenrui333** [comment]
yea, it probably does not support 8.0 at this point?

https://github.com/unfrl/dug/blob/main/cli/dug.csproj#L5

File: Formula/d/dug.rb [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+    os = OS.mac? ? "osx" : OS.kernel_name.downcase
+    arch = Hardware::CPU.intel? ? "x64" : Hardware::CPU.arch.to_s
+
+    args = %W[
+      --configuration Release
+      --framework net#{dotnet.version.major_minor}
+      --output #{libexec}
+      --runtime #{os}-#{arch}
+      --no-self-contained
+      -p:PublishSingleFile=true
```
Comment (chenrui333) [inline Formula/d/dug.rb:23]:
```suggestion
      -p:TargetFrameworks=net#{dotnet.version.major_minor}
      -p:Version=#{version}
      -p:PublishSingleFile=true
```

**Comment by chenrui333** [comment]
should be good now, just need upstream to mark 0.0.93 as latest release.

**Review by chenrui333 — APPROVED** [review]

**Comment by chenrui333** [comment]
@awfulwoman, thanks for your contribution to Homebrew! :tada: 🥇

Without awesome contributors like you, it would be impossible to maintain Homebrew to the high level of quality users have come to expect. Thank you!!!!

**Pull request merged**